assets/js/theme/category.js

```diff
@@ -10,13 +10,23 @@ export default class Category extends CatalogPage {
         this.validationDictionary = createTranslationDictionary(context);
     }
 
-    onReady() {
-        $('[data-button-type="add-cart"]').on('click', (e) => {
-            $(e.currentTarget).next().attr({
-                role: 'status',
-                'aria-live': 'polite',
-            });
+    setLiveRegionAttributes($element, roleType, ariaLiveStatus) {
+        $element.attr({
+            role: roleType,
+            'aria-live': ariaLiveStatus,
         });
+    }
+
+    onReady() {
+        $('[data-button-type="add-cart"]').on('click', (e) => this.setLiveRegionAttributes($(e.currentTarget).next(), 'status', 'polite'));
+
+        if ($('[data-shop-by-price]').length) {
+            if ($('.navList-action').hasClass('is-active')) {
+                $('a.navList-action.is-active').focus();
+            }
+
+            $('a.navList-action').on('click', () => this.setLiveRegionAttributes($('span.price-filter-message'), 'status', 'assertive'));
+        }
```

Comment:
Could we move these lines into separate method, because it's quite difficult to understand if you are not familiar with this logic

Reply:
done